modules/core/src/minmax.dispatch.cpp

```diff
@@ -457,10 +457,11 @@ static void reduceMinMax(cv::InputArray src, cv::OutputArray dst, ReduceMode mod
     CV_INSTRUMENT_REGION();
 
     cv::Mat srcMat = src.getMat();
-    axis = (axis + srcMat.dims) % srcMat.dims;
-    CV_Assert(srcMat.channels() == 1 && axis >= 0 && axis < srcMat.dims);
 
-    std::vector<int> sizes(srcMat.dims);
+    int dims = std::max(1, srcMat.dims);
+    axis = (axis + dims) % dims;
+    CV_Assert(srcMat.channels() == 1 && axis >= 0 && axis < dims);
+    std::vector<int> sizes(dims);
```

Comment:
Purpose of these change?

Reply:
I guess `srcMat.dims` equals zero causes numerical assertion at `axis = (axis + srcMat.dims) % srcMat.dims;`

Reply:
Not sure whether we should add this code in this patch. Seems like we need a test for this case and it worths a separate pull request.

Reply:
Please revert as the changes were merged in a different PR: https://github.com/opencv/opencv/pull/25487